### PR TITLE
Pydr 88 sql python driver

### PR DESCRIPTION
**Pull request by dlozina-macrometa**
## Description

We have added SQL language support and end-to-end test for that feature. Added minor refactoring and cleanup, updated some packages to set ourselves to do more later. 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

I did e2e test with test data that I included in our version control. 

## Reviews

Please identify two developers to review this change

- [ ] @ricardo-macrometa 
- [x] @koshy-thomas 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added new functional tests that prove my fix is effective or that my feature works
- [ ] Existing and new functional tests pass with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,77 +62,82 @@ Here is an overview example:
...
+demo_fabric = "_system"
+demo_user = "user@example.com"
+demo_user_name = "root"
+demo_collection = "employees"
+demo_stream = "demostream"
+collname = "employees"
+# --------------------------------------------------------------
+print("Create C8Client Connection...")
+client = C8Client(protocol='https', host=region, port=443,
+                  email=demo_tenant, password='hidden',
```
Comment (koshy-thomas) [inline README.md:81]: maybe we keep password='XXXXX' inline with our docs on website

File: README.md [hunk 2]
```diff
@@ -62,77 +62,82 @@ Here is an overview example:
...
+import time
+import warnings
+
+warnings.filterwarnings("ignore")
+region = "gdn1.macrometa.io"
+demo_tenant = "mytenant@example.com"
+demo_fabric = "_system"
+demo_user = "user@example.com"
+demo_user_name = "root"
+demo_collection = "employees"
```
Comment (koshy-thomas) [inline README.md:75]: demo_collection seems to be redundant

File: README.md [hunk 3]
```diff
@@ -62,77 +62,82 @@ Here is an overview example:
...
-    #--------------------------------------------------------------
-    print("Delete Collection...")
-    # Delete Collection
-    client.delete_collection(name=collname)
+from c8 import C8Client
+import time
+import warnings
+
+warnings.filterwarnings("ignore")
+region = "gdn1.macrometa.io"
```
Comment (koshy-thomas) [inline README.md:70]: we can make the region as "gdn.paas.macrometa.io" inline with our production url

File: c8/c8ql.py [hunk 4]
```diff
@@ -194,6 +196,8 @@ def execute(self,
         :param stream: Specify *true* and the query runs as a **stream**.
             The query result is not stored on server, but calculated on the fly.
         :type stream: bool
+        :param stream: Specify *true* and write sql query.
```
Comment (koshy-thomas) [inline c8/c8ql.py:199]: param name is "sql" here I guess, not stream

File: tests/e2e/test_sql.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+from conftest import response_content
+
+
+"""
+To run end to end test .env file in /e2e is needed.
+File needs to contain:
+FEDERATION_URL
+TENANT_EMAIL
+API_KEY
+FABRIC
```
[the remark this answers is not in the document]
Comment (dlozina-macrometa) [reply tests/e2e/test_sql.py:15]: Yes, sure I will do it! Thanks 👍

File: c8/c8ql.py (hunk 4 above)
Comment (dlozina-macrometa) [reply c8/c8ql.py:199]: 👍

File: README.md (hunk 3 above)
Comment (dlozina-macrometa) [reply README.md:70]: I agree 100%

File: README.md (hunk 1 above)
Comment (dlozina-macrometa) [reply README.md:81]: 👍

File: README.md (hunk 2 above)
Comment (dlozina-macrometa) [reply README.md:75]: 👍

File: README.md (hunk 1 above)
Comment (koshy-thomas) [reply README.md:87]: oops I guess you missed this part in the commit :)

Comment (koshy-thomas) [reply README.md:87]: the password thingy

File: c8/client.py [hunk 6]
```diff
@@ -110,17 +113,16 @@ def base_url(self):
...
         :type email: str | unicode
         :param password: Password for basic authentication.
         :type password: str | unicode
-        :param verify: Verify the connection by sending a test request.
-        :type verify: bool
+
         :returns: Standard fabric API wrapper.
-        :rtype: c8.fabric.StandardFabric
+        :type: c8.fabric.StandardFabric
         :raise c8.exceptions.ServerConnectionError: If **verify** was set
```
[the remark this answers is not in the document]
Comment (dlozina-macrometa) [reply c8/client.py:126]: 👍  top top

File: README.md [hunk 7]
```diff
@@ -149,17 +155,16 @@ Example for **real-time updates** from a collection in fabric:
...
   demo_user = "user@example.com"
   demo_user_name = "root"
-  demo_collection = "employees"
   demo_stream = "demostream"
   collname = "democollection"
   #--------------------------------------------------------------
   print("Create C8Client Connection...")
-  client = C8Client(protocol='https', host=region, port=443,
+  client = C8Client(protocol='https', host=federation, port=443,
                        email=demo_tenant, password='hidden',
```
Comment (koshy-thomas) [inline README.md:168]: here too we can change the password to password='XXXXX'.. but very minor

File: README.md [hunk 8]
```diff
@@ -184,13 +189,13 @@ Example to **publish** documents to a stream:
...
-  region = "gdn1.macrometa.io"
+  federation = "gdn.paas.macrometa.io"
   demo_tenant = "mytenant@example.com"
   demo_fabric = "_system"
   stream = "demostream"
   #--------------------------------------------------------------
   print("publish messages to stream...")
-  client = C8Client(protocol='https', host=region, port=443,
+  client = C8Client(protocol='https', host=federation, port=443,
                        email=demo_tenant, password='hidden',
```
Comment (koshy-thomas) [inline README.md:199]: +1 for password.. very minor thingy.. we can ignore for now if you want

File: README.md [hunk 9]
```diff
@@ -278,13 +283,13 @@ Advanced operations can be done using the `sream_colleciton` class.
...
-   region = "gdn1.macrometa.io"
+   federation = "gdn.paas.macrometa.io"
    demo_tenant = "mytenant@example.com"
    demo_fabric = "_system"
 
    #--------------------------------------------------------------
    print("consume messages from stream...")
-   client = C8Client(protocol='https', host=region, port=443)
+   client = C8Client(protocol='https', host=federation, port=443)
    demotenant = client.tenant(email=demo_tenant, password='hidden')
```
Comment (koshy-thomas) [inline README.md:293]: +1 for password.. very minor

File: README.md [hunk 10]
```diff
@@ -248,13 +253,13 @@ Example: **stream management**:
...
-  region = "gdn1.macrometa.io"
+  federation = "gdn.paas.macrometa.io"
   demo_tenant = "mytenant@example.com"
   demo_fabric = "_system"
   stream = "demostream"
   #--------------------------------------------------------------
   print("publish messages to stream...")
-  client = C8Client(protocol='https', host=region, port=443,
+  client = C8Client(protocol='https', host=federation, port=443,
                        email=demo_tenant, password='hidden',
```
Comment (koshy-thomas) [inline README.md:263]: +1 for password.. very minor

File: README.md [hunk 11]
```diff
@@ -215,13 +220,13 @@ Example to **subscribe** documents from a stream:
...
```
Comment (koshy-thomas) [inline README.md:230]: +1 for password.. very minor

**Review by koshy-thomas — APPROVED** [review]

File: README.md [hunk 12]
```diff
@@ -1,20 +1,21 @@
 # PyC8
 
-Welcome to the GitHub page for **pyC8**, a Python driver for the Digital Edge Fabric.
+Python SDK for the Macrometa Global Data Network.
 
-### Features
+# Compatibility
 
+- Python versions 3.4, 3.5 and 3.6 are supported.
```
Comment (ricardo-macrometa) [inline README.md:7]: I think we support up to 3.10 without issues.

**Review by ricardo-macrometa — CHANGES_REQUESTED** [review]

**Comment by ricardo-macrometa** [comment]
Also solve the merge conflicts.

File: README.md (hunk 12 above)
Comment (dlozina-macrometa) [reply README.md:7]: 👍  Yea, I have been testing with 3.10 as well, thanks

**Review by ricardo-macrometa — APPROVED** [review]

**Pull request merged**